From b35d4ec3d7e6a598253fc15f8238591951bd8845 Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Tue, 31 Dec 2019 18:40:24 +0800
Subject: [PATCH] media: rockchip: isp: remove RKISP_RK3326_USE_OLDMIPI

this is not used, remove it just for code simple.

Change-Id: I884d7cb086e0882dd7f4dd334edc1a8c3c883961
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rockchip/isp/capture.c | 12 -------
 drivers/media/platform/rockchip/isp/common.h  |  1 -
 drivers/media/platform/rockchip/isp/dev.c     | 22 -------------
 drivers/media/platform/rockchip/isp/rkisp.c   | 33 -------------------
 4 files changed, 68 deletions(-)

diff --git a/drivers/media/platform/rockchip/isp/capture.c b/drivers/media/platform/rockchip/isp/capture.c
index e986ff47072b..1eb140fa3c48 100644
--- a/drivers/media/platform/rockchip/isp/capture.c
+++ b/drivers/media/platform/rockchip/isp/capture.c
@@ -1244,12 +1244,8 @@ static int rkisp_start(struct rkisp_stream *stream)
 		return -EINVAL;
 	}
 
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13)
-#else
 	if (dev->isp_ver == ISP_V12 ||
 		dev->isp_ver == ISP_V13)
-#endif
 		raw_config_mi(stream);
 
 	if (stream->ops->set_data_path)
@@ -2106,12 +2102,8 @@ void rkisp_unregister_stream_vdevs(struct rkisp_device *dev)
 	if (dev->isp_ver != ISP_V10_1)
 		rkisp_unregister_stream_vdev(sp_stream);
 
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13)
-#else
 	if (dev->isp_ver == ISP_V12 ||
 		dev->isp_ver == ISP_V13)
-#endif
 		rkisp_unregister_stream_vdev(raw_stream);
 }
 
@@ -2135,12 +2127,8 @@ static int rkisp_register_stream_vdev(struct rkisp_stream *stream)
 		break;
 	case RKISP_STREAM_RAW:
 		vdev_name = RAW_VDEV_NAME;
-#if RKISP_RK3326_USE_OLDMIPI
-		if (dev->isp_ver != ISP_V13)
-#else
 		if (dev->isp_ver != ISP_V12 &&
 			dev->isp_ver != ISP_V13)
-#endif
 			return 0;
 		break;
 	default:
diff --git a/drivers/media/platform/rockchip/isp/common.h b/drivers/media/platform/rockchip/isp/common.h
index 8e784635ec6f..5b66bb9b9ff6 100644
--- a/drivers/media/platform/rockchip/isp/common.h
+++ b/drivers/media/platform/rockchip/isp/common.h
@@ -58,7 +58,6 @@
 
 #define RKISP_EMDDATA_FIFO_MAX		4
 #define RKISP_DMATX_CHECK              0xA5A5A5A5
-#define RKISP_RK3326_USE_OLDMIPI	0
 
 enum rkisp_sd_type {
 	RKISP_SD_SENSOR,
diff --git a/drivers/media/platform/rockchip/isp/dev.c b/drivers/media/platform/rockchip/isp/dev.c
index 4a8e7abd8610..8987b9e250ee 100644
--- a/drivers/media/platform/rockchip/isp/dev.c
+++ b/drivers/media/platform/rockchip/isp/dev.c
@@ -327,12 +327,8 @@ static int rkisp_create_links(struct rkisp_device *dev)
 	if (ret < 0)
 		return ret;
 
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13) {
-#else
 	if (dev->isp_ver == ISP_V12 ||
 		dev->isp_ver == ISP_V13) {
-#endif
 		/* MIPI RAW links */
 		source = &dev->isp_sdev.sd.entity;
 		sink = &dev->stream[RKISP_STREAM_RAW].vnode.vdev.entity;
@@ -635,12 +631,8 @@ static irqreturn_t rkisp_mipi_irq_hdl(int irq, void *ctx)
 	unsigned int mis_val;
 	unsigned int err1, err2, err3;
 
-#if RKISP_RK3326_USE_OLDMIPI
-	if (rkisp_dev->isp_ver == ISP_V13) {
-#else
 	if (rkisp_dev->isp_ver == ISP_V13 ||
 		rkisp_dev->isp_ver == ISP_V12) {
-#endif
 		err1 = readl(rkisp_dev->base_addr + CIF_ISP_CSI0_ERR1);
 		err2 = readl(rkisp_dev->base_addr + CIF_ISP_CSI0_ERR2);
 		err3 = readl(rkisp_dev->base_addr + CIF_ISP_CSI0_ERR3);
@@ -653,20 +645,6 @@ static irqreturn_t rkisp_mipi_irq_hdl(int irq, void *ctx)
 		mis_val = readl(rkisp_dev->base_addr + CIF_MIPI_MIS);
 		if (mis_val)
 			rkisp_mipi_isr(mis_val, rkisp_dev);
-
-		/*
-		 * As default interrupt mask for csi_rx are on,
-		 * when resetting isp, interrupt from csi_rx maybe arise,
-		 * we should clear them.
-		 */
-#if RKISP_RK3326_USE_OLDMIPI
-		if (rkisp_dev->isp_ver == ISP_V12) {
-			/* read error state register to clear interrupt state */
-			readl(rkisp_dev->base_addr + CIF_ISP_CSI0_ERR1);
-			readl(rkisp_dev->base_addr + CIF_ISP_CSI0_ERR2);
-			readl(rkisp_dev->base_addr + CIF_ISP_CSI0_ERR3);
-		}
-#endif
 	}
 
 	return IRQ_HANDLED;
diff --git a/drivers/media/platform/rockchip/isp/rkisp.c b/drivers/media/platform/rockchip/isp/rkisp.c
index a631725d6bcf..5b1ab2518ee5 100644
--- a/drivers/media/platform/rockchip/isp/rkisp.c
+++ b/drivers/media/platform/rockchip/isp/rkisp.c
@@ -228,11 +228,7 @@ static void rkisp_config_clk(struct rkisp_device *dev, int on)
 
 	writel(val, dev->base_addr + CIF_ICCL);
 
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13) {
-#else
 	if (dev->isp_ver == ISP_V12 || dev->isp_ver == ISP_V13) {
-#endif
 		val = !on ? 0 :
 		      CIF_CLK_CTRL_MI_Y12 | CIF_CLK_CTRL_MI_SP |
 		      CIF_CLK_CTRL_MI_RAW0 | CIF_CLK_CTRL_MI_RAW1 |
@@ -501,12 +497,8 @@ static int rkisp_config_mipi(struct rkisp_device *dev)
 		}
 	}
 
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13) {
-#else
 	if (dev->isp_ver == ISP_V13 ||
 		dev->isp_ver == ISP_V12) {
-#endif
 		/* lanes */
 		writel(lanes - 1, base + CIF_ISP_CSI0_CTRL1);
 
@@ -534,19 +526,6 @@ static int rkisp_config_mipi(struct rkisp_device *dev)
 			    CIF_MIPI_CTRL_SHUTDOWNLANES(0xf) |
 			    CIF_MIPI_CTRL_ERR_SOT_SYNC_HS_SKIP |
 			    CIF_MIPI_CTRL_CLOCKLANE_ENA;
-
-#if RKISP_RK3326_USE_OLDMIPI
-		if (dev->isp_ver == ISP_V12) {
-			writel(0, base + CIF_ISP_CSI0_CTRL0);
-			writel(0, base + CIF_ISP_CSI0_MASK1);
-			writel(0, base + CIF_ISP_CSI0_MASK2);
-			writel(0, base + CIF_ISP_CSI0_MASK3);
-			/* clear interrupts state */
-			readl(base + CIF_ISP_CSI0_ERR1);
-			readl(base + CIF_ISP_CSI0_ERR2);
-			readl(base + CIF_ISP_CSI0_ERR3);
-		}
-#endif
 		writel(mipi_ctrl, base + CIF_MIPI_CTRL);
 
 		/* Configure Data Type and Virtual Channel */
@@ -711,11 +690,7 @@ static int rkisp_isp_stop(struct rkisp_device *dev)
 	 * Stop ISP(isp) ->wait for ISP isp off
 	 */
 	/* stop and clear MI, MIPI, and ISP interrupts */
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13) {
-#else
 	if (dev->isp_ver == ISP_V12 || dev->isp_ver == ISP_V13) {
-#endif
 		writel(0, base + CIF_ISP_CSI0_MASK1);
 		writel(0, base + CIF_ISP_CSI0_MASK2);
 		writel(0, base + CIF_ISP_CSI0_MASK3);
@@ -732,11 +707,7 @@ static int rkisp_isp_stop(struct rkisp_device *dev)
 
 	writel(0, base + CIF_MI_IMSC);
 	writel(~0, base + CIF_MI_ICR);
-#if RKISP_RK3326_USE_OLDMIPI
-	if (dev->isp_ver == ISP_V13) {
-#else
 	if (dev->isp_ver == ISP_V12 || dev->isp_ver == ISP_V13) {
-#endif
 		writel(0, base + CIF_ISP_CSI0_CTRL0);
 	} else {
 		val = readl(base + CIF_MIPI_CTRL);
@@ -825,11 +796,7 @@ static int rkisp_isp_start(struct rkisp_device *dev)
 
 	/* Activate MIPI */
 	if (sensor && sensor->mbus.type == V4L2_MBUS_CSI2) {
-#if RKISP_RK3326_USE_OLDMIPI
-		if (dev->isp_ver == ISP_V13) {
-#else
 		if (dev->isp_ver == ISP_V12 || dev->isp_ver == ISP_V13) {
-#endif
 			/* clear interrupts state */
 			readl(base + CIF_ISP_CSI0_ERR1);
 			readl(base + CIF_ISP_CSI0_ERR2);
-- 
2.35.3

